importer/api/__init__.py

```diff
@@ -109,7 +108,8 @@ def import_course_from_path(path, repo_id, user_id):
     )
     bundle.import_from_directory(path)
     static_dir = join(path, 'static')
-    course = import_course(bundle, repo_id, user_id, static_dir)
+    with transaction.atomic():
+        course = import_course(bundle, repo_id, user_id, static_dir)
```

Comment:
I don't believe there would be any extra memory usage, I think transactions do their business in a separate blocks in the database and then switch over once the transaction is complete. I talked to @bdero and he was thinking that since we're using Postgres the bad experiences with transactions with edx may not apply.